pkg/imgutil/commit/commit.go

```diff
@@ -173,6 +191,34 @@ func Commit(ctx context.Context, client *containerd.Client, container containerd
 	return configDigest, nil
 }
 
+// ensureManifestLayers checks layers specified in manifest, reconstruct contents if it is missing and change manifest
+func ensureManifestLayers(ctx context.Context, manifest *ocispec.Manifest, cs content.Store, config *ocispec.Image, sn snapshots.Snapshotter, comparer diff.Comparer) error {
+	if len(manifest.Layers)+1 != len(config.RootFS.DiffIDs) {
+		return fmt.Errorf("layer size in manifest(%v)+1 does not match with config(%v)", len(manifest.Layers), len(config.RootFS.DiffIDs))
+	}
+	for i, layer := range manifest.Layers {
+		_, err := cs.Info(ctx, layer.Digest)
+		if err == nil {
+			continue
+		}
+		if !errdefs.IsNotFound(err) {
+			return fmt.Errorf("get layer %v failed: %w", layer.Digest.String(), err)
+		}
+		snapshotID := identity.ChainID(config.RootFS.DiffIDs[:i+1]).String()
```

Comment:
Can we get the snapshot ID from the metadata store (e.g. `containers.Container`) rather than manually computing it here?

Reply:
yea that works too. My concern is that since we are reconstructing layers (digest recorded in manifest is not found), we better rely on manifest/config which is image relevant instead of snapshotter plugin..